internal/tools/update-go-azure-sdk/main.go

```diff
@@ -0,0 +1,729 @@
+			if !ok {
+				existing = make([]string, 0)
+			}
+			containsExistingValue := false
+			for _, val := range existing {
+				if val == apiVersion {
+					containsExistingValue = true
+					break
+				}
+			}
+			if !containsExistingValue {
+				existing = append(existing, apiVersion)
+			}
+
+			out.newServicesToApiVersions[serviceName] = existing
+
+			continue
+		}
+	}
+	return out
+}
+
+func updateVersionOfGoAzureSDK(ctx context.Context, workingDirectory, newApiVersion string) error {
+	logger.Debug(fmt.Sprintf("Updating the version of `go-azure-sdk` to %q", newApiVersion))
+	args := []string{
+		"get",
+		fmt.Sprintf("github.com/hashicorp/go-azure-sdk@%s", newApiVersion),
+	}
+	cmd := exec.CommandContext(ctx, "go", args...)
+	cmd.Dir = workingDirectory
+	_ = cmd.Start()
+	_ = cmd.Wait()
+	logger.Debug(fmt.Sprintf("Updated the version of `go-azure-sdk` to %q", newApiVersion))
+
+	logger.Debug("Vendoring the changes..")
+	goModTidyAndVendor(ctx, workingDirectory)
+
+	logger.Debug("Committing the changes..")
+	message := fmt.Sprintf("dependencies: updating to version `%s` of `github.com/hashicorp/go-azure-sdk`", newApiVersion)
+	if err := stageAndCommitChanges(workingDirectory, message); err != nil {
+		return fmt.Errorf("staging/committing changes to %q: %+v", workingDirectory, err)
+	}
+	return nil
+}
+
+func goModTidyAndVendor(ctx context.Context, workingDirectory string) {
+	logger.Debug(fmt.Sprintf("Running `go mod tidy` in %q..", workingDirectory))
+	tidyArgs := []string{
+		"mod",
+		"tidy",
+	}
+	tidyCmd := exec.CommandContext(ctx, "go", tidyArgs...)
+	tidyCmd.Dir = workingDirectory
+	_ = tidyCmd.Start()
+	_ = tidyCmd.Wait()
+	logger.Debug(fmt.Sprintf("Run `go mod tidy` in %q.", workingDirectory))
+
+	logger.Debug(fmt.Sprintf("Running `go mod vendor` in %q..", workingDirectory))
+	vendorArgs := []string{
+		"mod",
+		"vendor",
+	}
+	vendorCmd := exec.CommandContext(ctx, "go", vendorArgs...)
+	vendorCmd.Dir = workingDirectory
+	_ = vendorCmd.Start()
+	_ = vendorCmd.Wait()
+	logger.Debug(fmt.Sprintf("Run `go mod vendor` in %q.", workingDirectory))
+	return
+}
+
+func stageAndCommitChanges(workingDirectory string, message string) error {
+	repo, err := git.PlainOpen(workingDirectory)
+	if err != nil {
+		return fmt.Errorf("opening %q: %+v", workingDirectory, err)
+	}
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return fmt.Errorf("opening worktree: %+v", err)
+	}
+
+	// Whilst `Commit` has an add below it doesn't capture all changes, hence we need
+	// to explicitly stage all changes (including the vendored changes)
+	logger.Trace(fmt.Sprintf("Staging all changes in %q..", workingDirectory))
+	addOpts := &git.AddOptions{
+		All:  true,
+		Glob: "**",
+	}
+	if err := worktree.AddWithOptions(addOpts); err != nil {
+		return fmt.Errorf("staging changes: %+v", err)
+	}
+
+	logger.Trace(fmt.Sprintf("Committing all changes in %q..", workingDirectory))
+	opts := &git.CommitOptions{
+		All: true,
```

Comment:
Likewise, perhaps `All` not needed here if the index was already populated above?